sycl/source/detail/config.hpp

```diff
@@ -163,6 +165,32 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_DEVICE_TRIPLES> {
+  using BaseT = SYCLConfigBase<SYCL_DEVICE_TRIPLES>;
+
+public:
+  static device_triple_list *get() {
+    static bool Initialized = false;
+    static device_triple_list *TripleList = nullptr;
+
+    // Configuration parameters are processed only once, like reading a string
+    // from environment and converting it into a typed object.
+    if (Initialized) {
```

Comment:
```suggestion
    if (!TripleList) {
```
This should eliminate the need of `Initialized` var. BTW BaseT::getRawValue already caches the value.

Reply:
I think the boolean is helpful to avoid checking the env var over and over again. I assume once env var is checked, if no env var is set, we can immediately return nullptr without checking env var again.

Reply:
Checking env var(return value of `getRawValue`) should be as easy as checking bool var.
This is minor comment, so feel free to ignore it.